File: 4/CalcOrdenSuperior.scala


object CalcOrdenSuperior {
  def main(args: Array[String]) {

    println(operacion(10, 2, suma))
    println(operacion(10, 2, resta))
    println(operacion(10, 2, multi))
    println(operacion(10, 2, division))

  }
  def suma(x: Int, y: Int): Int = x + y
  def resta(x: Int, y: Int): Int = x - y
  def multi(x: Int, y: Int): Int = x * y
  def division(x: Int, y: Int): Int = x / y

  def operacion(x: Int, y: Int, fun: (Int, Int) => Int): Int = fun(x, y)

}

File: 4/CalcOrdenSuperior.py

def suma(x, y): return x + y
def resta(x, y): return x - y
def multi(x, y): return x * y
def division(x, y): return x / y

def operacion(x,y, fun):
  return fun(x,y)

print(operacion(10,2,suma))
print(operacion(10,2,resta))
print(operacion(10,2,multi))
print(operacion(10,2,division))